### PR TITLE
Add command line input for tag management

### PR DESCRIPTION
**Pull request by andreaTP**
Added command line arguments to pass default parameter on tagRelease Step.

### CODE REVIEW

**Comment by andreaTP** [comment]
maybe @jroper @gseitz or anybody else can comment here?

**Comment by jroper** [comment]
I'm not 100% sure of what the use case is here, nor do I 100% understand how the `tag-default` attribute is used, could you explain a little more?

File: src/sbt-test/sbt-release/tag-default/test [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+$ exec git add .
+$ exec git commit -m init
+$ exec git tag v0.1.0
+
+-> release with-defaults
+-> release with-defaults tag-default a
+
+> release with-defaults tag-default o
+> release with-defaults tag-default k
+> release with-defaults tag-default v0.1.0-debug
```
Comment (jroper) [inline src/sbt-test/sbt-release/tag-default/test]: I don't understand this scripted tests - why does `release with defaults` fail and `release with-defaults tag-default a` also fail, but `release with-defaults tag-default o` doesn't fail?  What's so special about `o` that makes it different to `a`?  I just don't get it.  Could you add some comments here to explain what's being tested and what's going on?

File: src/main/scala/ReleasePlugin.scala [hunk 2]
```diff
@@ -136,17 +137,21 @@ object ReleasePlugin extends AutoPlugin {
         (Space ~> token("release-version") ~> Space ~> token(StringBasic, "<release version>")) map ParseResult.ReleaseVersion
       private[this] val NextVersion: Parser[ParseResult] =
         (Space ~> token("next-version") ~> Space ~> token(StringBasic, "<next version>")) map ParseResult.NextVersion
+      private[this] val TagDefault: Parser[ParseResult] =
+        (Space ~> token("tag-default") ~> Space ~> token(StringBasic, "<next version>")) map ParseResult.TagDefault
```
Comment (xuwei-k) [inline src/main/scala/ReleasePlugin.scala]:
```diff
-      (Space ~> token("tag-default") ~> Space ~> token(StringBasic, "<next version>")) map ParseResult.TagDefault
+      (Space ~> token("tag-default") ~> Space ~> token(StringBasic, "<tag default>")) map ParseResult.TagDefault
```

Comment (andreaTP) [inline src/main/scala/ReleasePlugin.scala]: right thanks

**Comment by andreaTP** [comment]
**TL;DR**
Just let user adjust hardcoded default of the tagging step

**Rationale**
The usage of this parameter is explicitly for usage on CI, ```release with-default``` alone doesn't let me decide the rules for tagging explicitly, and the only ```default``` is the hardcoded ```abort```.
With this PR you will be able to pass through command line arguments different tagging default.
My specific use case is that I want to overwrite SNAPSHOT tags on git, but this enable also the user to give personal tags through the argument in CI.

File: src/sbt-test/sbt-release/tag-default/test (hunk 1 above)
Comment (andreaTP) [inline src/sbt-test/sbt-release/tag-default/test]: going to add comments around.

**Comment by andreaTP** [comment]
done fixes and addressed comments.
rebased modifications.

Thanks for taking care and for comments :-)

**Comment by andreaTP** [comment]
CI failure looks spurious to me

**Comment by jroper** [comment]
Ah, I get it now.  Perhaps then a more descriptive name would be `default-tag-exists-answer`. Also, the description of the value used in the parser should probably be `o|k|a`, rather than `<tag-default>`.

Also, this needs to be documented in the `README`, eg, in the section about `with-defaults`, saying that the answer to the existing VCS tag prompt can be overridden using the `default-tag-exists-answer` argument.

**Comment by andreaTP** [comment]
Done, thanks a lot for the feedback! :-)

**Comment by andreaTP** [comment]
Thanks!

**Comment by andreaTP** [comment]
Due to recent updates, could you please publish a release?
Thanks in advance
